Extensions/PrimitiveDrawing/Extension.cpp

```diff
@@ -168,9 +187,194 @@ void DeclarePrimitiveDrawingExtension(gd::PlatformExtension& extension) {
+                _("Draw from _PARAM1_;_PARAM2_ to _PARAM5_;_PARAM6_ a quadratic curve (control point: _PARAM3_;_PARAM4_) "
+                "with _PARAM0_"),
+                _("Drawing"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("X position of start point"))
+      .AddParameter("expression", _("Y position of start point"))
+      .AddParameter("expression", _("Control point x"))
+      .AddParameter("expression", _("Control point y"))
+      .AddParameter("expression", _("Destination point x"))
+      .AddParameter("expression", _("Destination point y"))
+      .SetFunctionName("drawQuadraticCurve")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("BeginFillPath",
+                _("Begin fill path"),
+                _("Begins drawing filling of an advanced path"),
+                _("Begins drawing filling of an advanced path "
+                  " with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("Start drawing x"))
+      .AddParameter("expression", _("Start drawing y"))
+      .SetFunctionName("beginFillPath")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("EndFillPath",
+                _("End fill path"),
+                _("Finish the filling drawing in an advanced path"),
+                _("Finish the filling drawing in an advanced path "
+                  " with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/delete24.png",
+                "res/actions/delete.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .SetFunctionName("endFillPath")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("MovePathTo",
+                _("Move path to"),
+                _("Move the origin of a path."),
+                _("Move the origin of a path to _PARAM1_;_PARAM2_ "
+                  " with _PARAM0_"),
+                _("Advanced"),
+                "res/actions/position24.png",
+                "res/actions/position.png")
+
+      .AddParameter("object", _("Shape Painter object"), "Drawer")
+      .AddParameter("expression", _("X position of start point"))
+      .AddParameter("expression", _("Y position of start point"))
+      .SetFunctionName("drawPathMoveTo")
+      .SetIncludeFile("PrimitiveDrawing/ShapePainterObject.h");
+
+  obj.AddAction("PathLineTo",
+                _("Path line to"),
+                _("Add to a path a line to a position, the origin come from the previous action or from BeginFillPath or PathMoveTo, by default if nothing exist, start position will be the object's position."),
```

Comment:
Better use shorter sentences. Also, use the full, "user friendly" name of the actions (i.e: the one shown in the UI) in the descriptions:

`Add to a path a line to a position. The origin comes from the previous action or from "Begin fill path" or "Move path drawing position". By default if nothing exist, start position will be the object's position.`